### PR TITLE
Enable header toolbar buttons to be disabled and use it for Add an Order

### LINKED ISSUE #19396: Multistore - Order > Order - Disable add new button in all shops or in a group context and add message on hover
Currently, the button  “Add new order” is enabled in all shops or in a group context and a message is diplayed when you click on it "You have to select a shop before creating new orders."

To avoid frustration, we should inform the merchants earlier

As specified in the [multistore specials cases specs](https://github.com/PrestaShop/prestashop-specs/blob/master/broader-topics/multistorespecialsspecs.md#list-of-pages-for-specific-case-1-add-new-actions-only-enabled-in-a-specific-shop-context), the “Add new order” action is not possible in all shops or in a group context, so the button should be disabled in these contexts and we should display a message when the user hovers the "add new" button: "You can use this feature in a single shop context only. Switch context to enable it."

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Enable header toolbar buttons to be disabled and use it for Add an Order. See below for details
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/19396
| How to test?      | See https://github.com/PrestaShop/PrestaShop/issues/19396
| Possible impacts? | I modified 1 SCSS file but it's only used for header toolbar buttons. I modified how the "Add an order" button is rendered when shop context is not a "single shop" context.

### Goal

This PR implements what has been specified in https://github.com/PrestaShop/PrestaShop/issues/19396 . Display has been validated by @marionf .

Capture d’écran 2021-04-06 à 18 11 45

### Code modifications

I modified the twig rendering for header toolbar buttons so I can disable them.

I added some business logic inside the OrderController to configure properly the header toolbar button "Add an order" when shop context is not a "single shop" context. All configuration settings use the regular header toolbar buttons configuration settings (with the new ones I introduced).

## Details about the rendering

I used Bootstrap [disabled](https://www.w3schools.com/tags/att_button_disabled.asp) behavior and [tooltip](https://www.w3schools.com/bootstrap/bootstrap_tooltip.asp) behaviors.

However Bootstrap buttons or link-that-act-as-buttons cannot have a hover message ⚠️  . The hover message is disabled. This is why I introduced a `auto-pointer-events` CSS class that allows me to bypass this native behavior and allows me to display the hover message.

Additionnally, I replace the `href` by `#` because it's [impossible to disable links](https://css-tricks.com/how-to-disable-links/). So I make the link an anchor to make it "useless".

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23907)

### CODE REVIEW

File: admin-dev/themes/new-theme/scss/components/_button.scss [hunk 1]
```diff
@@ -13,3 +13,7 @@
     }
   }
 }
+
+a.btn.auto-pointer-events {
```
Comment (matks) [inline admin-dev/themes/new-theme/scss/components/_button.scss:17]: Explanation about this CSS class in the PR body

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 2]
```diff
@@ -163,12 +163,23 @@ private function getOrderToolbarButtons(): array
...
             'icon' => 'add_circle_outline',
+            'disabled' => !$isSingleShopContext,
         ];
 
+        if (!$isSingleShopContext) {
+            $toolbarButtons['add']['help'] = $this->trans(
+                'You can use this feature in a single shop context only. Switch context to enable it.',
+                'Admin.Orderscustomers.Feature'
+            );
+            $toolbarButtons['add']['href'] = '#';
```
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php:180]: Explanation about this trick in the PR body

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 3]
```diff
@@ -486,10 +497,10 @@ public function viewAction(int $orderId, Request $request): Response
...
             $this->dispatchHook(
                 'actionGetAdminOrderButtons', [
-                    'controller' => $this,
-                    'id_order' => $orderId,
-                    'actions_bar_buttons_collection' => $backOfficeOrderButtons,
-                ]);
+                'controller' => $this,
+                'id_order' => $orderId,
+                'actions_bar_buttons_collection' => $backOfficeOrderButtons,
+            ]);
```
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]: My IDE insisted on refactoring that. Even when I put back the original content it AUTOMATICALLY fixes the indent 😆

**Review by matthieu-rolland — DISMISSED** [review]

File: admin-dev/themes/new-theme/template/page_header_toolbar.tpl [hunk 4]
```diff
@@ -35,7 +35,7 @@
               {if $k != 'back' && $k != 'modules-list'}
                 {* TODO: REFACTOR ALL THIS THINGS *}
                 <a
-                  class="btn btn-primary {if isset($btn.target) && $btn.target} _blank{/if} pointer"{if isset($btn.href)}
+                  class="btn btn-primary {if isset($btn.target) && $btn.target} _blank{/if} {if isset($btn.disabled) && $btn.disabled} disabled auto-pointer-events{/if} pointer"{if isset($btn.href)}
```
Comment (Progi1984) [inline admin-dev/themes/new-theme/template/page_header_toolbar.tpl]:
```suggestion
                  class="btn btn-primary{if isset($btn.target) && $btn.target} _blank{/if}{if isset($btn.disabled) && $btn.disabled} disabled auto-pointer-events{/if} pointer"{if isset($btn.href)}
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 5]
```diff
@@ -486,10 +497,10 @@ public function viewAction(int $orderId, Request $request): Response
         try {
             $this->dispatchHook(
                 'actionGetAdminOrderButtons', [
```
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
                'actionGetAdminOrderButtons',
                [
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php (hunk 3 above)
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
                    'controller' => $this,
                    'id_order' => $orderId,
                    'actions_bar_buttons_collection' => $backOfficeOrderButtons,
                ]
            );
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Review by Progi1984 — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Comment by sowbiba** [comment]
Approved. Waiting for @matthieu-rolland review to push to QA

**Comment by matks** [comment]
> Approved. Waiting for @matthieu-rolland review to push to QA

I've been doing this PR while discussing with him 😄

**Comment by khouloudbelguith** [comment]
Hi @matks,

It is ok :heavy_check_mark: 
- Create Order from the BO when multistore context disabled
- Create Order in different shops with different Group via desktop or mobile
- When "All shops" is selected or Group is selected => Add new Order is disabled
 
https://drive.google.com/file/d/19z58prFBaxHJGm96eH4uEhsBZSzPvEqR/view?usp=sharing

Thank you!

**Comment by matks** [comment]
Thank you @khouloudbelguith !

**Pull request merged**